core/text/text_and_dictation.py

```diff
@@ -85,24 +85,15 @@ def prose_percent(m) -> str:
     return s + "%"
 
 
-@mod.capture(rule="<user.number_string> {user.currency_denomination}")
-def prose_simple_money(m) -> str:
-    return m.currency_denomination + m.number_string
-
-
-@mod.capture(
-    rule="<user.number_string> {user.currency_denomination} [and] <user.number_string> [cents|pence]"
-)
-def prose_money_with_cents(m) -> str:
-    return m.currency_denomination + m.number_string_1 + "." + m.number_string_2
-
-
-@mod.capture(rule="<user.prose_money_with_cents> | <user.prose_simple_money>")
+@mod.capture(rule="<user.number_string> {user.currency_denomination} [[and] <user.number_string> [cents|pence]]")
```

Comment:
Huh. I tried exactly this rule and it didn't seem to work, I thought maybe nested []'s were not supported.  I'm assuming you tested it, so I must have done something else wrong.

Reply:
It seemed to work in my testing, yeah. Feel free to test it on your end, maybe I missed something?

Nested [] should definitely be supported, though.